File: Client/iOS/Component/ToolKit/Common/ComponentProtocol/LiveAddrProtocol/LiveAddrProtocol.m
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0
//

#import "LiveAddrProtocol.h"

@implementation LiveAddrProtocol

- (void)getRTMPAddr:(NSString *)taskID
              block:(void (^)(NSString *addr))block {
    NSObject<LiveAddrDelegate> *delegate = [[NSClassFromString(@"RTMPGenerator") alloc] init];

    if ([delegate respondsToSelector:@selector(protocol:getRTMPAddr:block:)]) {
        [delegate protocol:self getRTMPAddr:taskID block:block];
    }
}



@end

File: Client/iOS/Component/RTCAPIExample/RTCAPIExample/Common/LiveAddrProtocol.swift
// Copyright (c) 2024 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0
//

import Foundation

@objc 
public protocol LiveAddrProtocol : NSObjectProtocol {
    static func generate(_ taskId: String) -> String
}

func generateLiveAddr(taskId: String) -> String{
    guard let clazz = NSClassFromString("RTMPGenerator") as? LiveAddrProtocol.Type else { return "" }
    return clazz.generate(taskId)
}
